eq-author-api/app.js

```diff
@@ -20,38 +19,50 @@ const pino = pinoMiddleware();
 const logger = createLogger(pino.logger);
 addErrorLoggingToSchema(schema, logger);
 
-const knex = require("./db");
+const db = require("./db");
 
-const context = { repositories: repositories(knex) };
+db(process.env.DB_SECRET_ID)
+  .then(async conf => {
+    let knex = require("knex")(conf);
+    logger.log("Running Migrate");
+    await knex.migrate.latest();
+    logger.log("Ran Migrate");
 
-app.use(
-  "/graphql",
-  pino,
-  cors(),
-  bodyParser.json(),
-  graphqlExpress({
-    schema,
-    context,
-    formatError: logger.log
+    const context = { repositories: repositories(knex) };
+
+    app.use(
+      "/graphql",
+      pino,
+      cors(),
+      bodyParser.json(),
+      graphqlExpress({
+        schema,
+        context,
+        formatError: logger.log
+      })
+    );
+
+    app.get("/status", status);
+
+    app.get("/launch/:questionnaireId", getLaunchUrl(context));
+    if (process.env.NODE_ENV === "development") {
+      app.post("/import", bodyParser.json({ limit: "50mb" }), importAction);
+    }
+
+    if (process.env.NODE_ENV === "development") {
+      app.use(
+        "/graphiql",
+        pino,
+        cors(),
+        graphiqlExpress({ endpointURL: "/graphql" })
+      );
+    }
+
+    app.listen(PORT, "0.0.0.0", () => {
+      logger.log("Listening on port", PORT);
```

Comment:
It doesn't seem to log the `PORT` so the api to our previous logging is different.